src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SqlDataReaderSmi.cs

```diff
@@ -1347,36 +1338,36 @@ internal override void MetaDataAvailable(SmiQueryMetaData[] md, bool nextEventIs
             // Obsolete V2- method
             internal override void RowAvailable(ITypedGetters row)
             {
-                if (Bid.AdvancedOn)
+                if (Log.IsTraceEnabled())
                 {
-                    Bid.Trace("<sc.SqlDataReaderSmi.ReaderEventSink.RowAvailable|ADV> %d# (v2).\n", reader.ObjectID);
+                    Log.Trace($"<sc.SqlDataReaderSmi.ReaderEventSink.RowAvailable|ADV> {reader.ObjectID}# (v2).");
                 }
                 this.reader.RowAvailable(row);
             }
 
             internal override void RowAvailable(ITypedGettersV3 row)
             {
-                if (Bid.AdvancedOn)
+                if (Log.IsTraceEnabled())
                 {
-                    Bid.Trace("<sc.SqlDataReaderSmi.ReaderEventSink.RowAvailable|ADV> %d# (ITypedGettersV3).\n", reader.ObjectID);
+                    Log.Trace($"<sc.SqlDataReaderSmi.ReaderEventSink.RowAvailable|ADV> {reader.ObjectID}# (ITypedGettersV3).");
                 }
                 this.reader.RowAvailable(row);
             }
 
             internal override void RowAvailable(SmiTypedGetterSetter rowData)
             {
-                if (Bid.AdvancedOn)
```

Comment:
`Bid.AdvancedOn()` is not the same as regular Bid.trace logs, and should not be treated same. It may lead to unwanted lengthly logs and should only be enabled on request.

What are the keyword options we have to enable these advanced logs given current setup?

Reply:
I have added a field in EventSource for AdvanceTrace. It checks for conditions of the enabled events   `Log.IsEnabled(EventLevel.LogAlways, EventKeywords.All)`. This only Logs when no filtering is requested.